ddsrouter_core/src/cpp/participant/implementations/rtps/DiscoveryServerParticipant.cpp

```diff
@@ -91,16 +91,40 @@ DiscoveryServerParticipant::participant_attributes_(
             {
                 has_listening_tcp_ipv4 = true;
 
-                std::shared_ptr<eprosima::fastdds::rtps::TCPv4TransportDescriptor> descriptor =
-                        std::make_shared<eprosima::fastdds::rtps::TCPv4TransportDescriptor>();
+                std::shared_ptr<eprosima::fastdds::rtps::TCPv4TransportDescriptor> descriptor;
+                bool same_wan_addr = false;
 
-                descriptor->add_listener_port(address.port());
-                descriptor->set_WAN_address(address.ip());
+                auto it = params.userTransports.begin();
+                while (it != params.userTransports.end())
+                {
+                    std::shared_ptr<eprosima::fastdds::rtps::TCPv4TransportDescriptor> tmp_descriptor =
+                            std::dynamic_pointer_cast<eprosima::fastdds::rtps::TCPv4TransportDescriptor>(*it);
+
+                    if ((tmp_descriptor != nullptr) && (address.ip() == tmp_descriptor->get_WAN_address()))
+                    {
+                        descriptor = tmp_descriptor;
+                        same_wan_addr = true;
+                        params.userTransports.erase(it);
```

Comment:
Comment this lines:
`Erasing old transport because it will be added afterwards` etc.